### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by joredudiaz**
# Summary
- authentication implemented
# Trello Card
https://trello.com/c/9P3SBbQH

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,10 +1,12 @@
 class User < ApplicationRecord
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
-  devise :database_authenticatable, :registerable,
-         :recoverable, :rememberable, :validatable
+  devise :database_authenticatable, :registerable, :recoverable, :rememberable, :trackable,
+         :validatable
+
+  include DeviseTokenAuth::Concerns::User
```
Comment (leoncalermo) [inline app/models/user.rb]: Move this include to the first line inside  the class, it's important to keep an order

File: db/migrate/20181025153846_devise_token_auth_create_users.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
+class DeviseTokenAuthCreateUsers < ActiveRecord::Migration[5.2]
```
Comment (leoncalermo) [inline db/migrate/20181025153846_devise_token_auth_create_users.rb:1]: This wasn't in the last pr too 🤔  ?

**Review by leoncalermo — COMMENTED** [review]
Fix these and its ready to merge 😉

File: app/models/user.rb (hunk 1 above)
Comment (joredudiaz) [reply app/models/user.rb]: Got it! I moved the include.

File: db/migrate/20181025153846_devise_token_auth_create_users.rb (hunk 2 above)
Comment (joredudiaz) [reply db/migrate/20181025153846_devise_token_auth_create_users.rb:1]: nup, the devise_token_auth gem was install in this branch and when I executed the installation It replaced the user migration for this. So this is "new".

File: db/migrate/20181025153846_devise_token_auth_create_users.rb [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      t.string   :last_sign_in_ip
+
+      ## Confirmable
+      t.string   :confirmation_token
+      t.datetime :confirmed_at
+      t.datetime :confirmation_sent_at
+      t.string   :unconfirmed_email # Only if using reconfirmable
+
+      ## Lockable
+      # t.integer  :failed_attempts, :default => 0, :null => false # Only if lock strategy is :failed_attempts
```
Comment (holywyvern) [inline db/migrate/20181025153846_devise_token_auth_create_users.rb]: Please, remove commented code

File: db/migrate/20181025153846_devise_token_auth_create_users.rb (hunk 2 above)
Comment (leoncalermo) [reply db/migrate/20181025153846_devise_token_auth_create_users.rb:1]: wait, so you updated a migration file?

**Review by leoncalermo — APPROVED** [review]

**Pull request merged**